### PR TITLE
Add Functional Item View Using NEXT/Router

### PR DESCRIPTION
**Pull request by garma-a**
This pull request introduces the ItemView component, a key UI element for managing item details within our application. The component integrates Material-UI (MUI) for styling and Next.js for routing, providing a comprehensive solution for displaying and interacting with item data.

Update Functionalities:

Importance: Allows users to set an item's importance status via a dropdown selector (BasicSelect). The selection updates the server using a PATCH request to /api/update_important/<id>.
Status: Provides a button to toggle the item’s operational status between open and close. Status updates are sent to the server with a PATCH request to /api/update_status/<id>.
Delete Operation: Features a confirmation dialog to prevent accidental deletions. Upon confirmation, the item is removed from the server via a DELETE request to /api/delete/<id>, and the user is navigated to the home page using Next.js’s useRouter for routing.

Styling and UX Enhancements: Utilizes Material-UI’s components and custom styles to ensure a modern, responsive design. Includes hover states and custom color schemes for a cohesive user experience.

Error Handling: Implements error handling for network requests to ensure robustness and provide user feedback in case of errors during updates or deletions.

Changes Made
Added the ItemView component with functionality for viewing, updating, and deleting items.
Integrated API interactions for updating item status and importance.
Implemented navigation and error handling features.

### CODE REVIEW

File: front/src/components/ItemView.tsx [hunk 1]
```diff
@@ -146,34 +147,67 @@ interface ItemViewProps {
...
-      important: value === 'true',
-    } : null);
+  const [open, setOpen] = React.useState(false);
+ 
+  const router = useRouter();
+  const handleImportantChange = async (value: string) => {
+  
+     if (row) {
+       try {
+         await fetch(`/api/update_important/${row.id}`, {
```
Comment (DevAbdoTolba) [inline front/src/components/ItemView.tsx]: update api to the new endpoint 
`/api/patch/update_important/<id>`

File: front/src/components/ItemView.tsx [hunk 2]
```diff
@@ -146,34 +147,67 @@ interface ItemViewProps {
...
-    setRow((prevRow) => prevRow ? {
-      ...prevRow,
-      status: status,
-    } : null);
+  const setStatus = async(status: string) => {
+     if (row) {
+       try {
+         await fetch(`/api/update_status/${row.id}`, {
+           method: "PATCH",
+           headers: { "Content-Type": "application/json" },
```
Comment (DevAbdoTolba) [inline front/src/components/ItemView.tsx:177]: update api to the new endpoint 
`/api/patch/update_status/<id>`

Comment (DevAbdoTolba) [inline front/src/components/ItemView.tsx:177]: update api to the new endpoint 
`/api/patch/update_status/<id>`

**Review by DevAbdoTolba — CHANGES_REQUESTED** [review]
Great working as charm 👏.
some changes are required 

- [x] No files more than the requested only (There are 4 uploaded, 1 is requested)
- [x] No updating or downloading new dependencies
- [x] And more about code requested below

File: front/src/components/ItemView.tsx [hunk 3]
```diff
@@ -190,11 +190,11 @@ const ItemView: React.FC<ItemViewProps> = ({ data }) => {
   const deleteItem = async() => {
      if (row) {
        try {
-         await fetch(`/api/delete/${row.id}`, {
+         await fetch(`/api/patch/delete/${row.id}`, {
```
Comment (DevAbdoTolba) [inline front/src/components/ItemView.tsx]: delete endpoitn is not updated. please revert to back to 
`/api/delete/<id>`

**Comment by garma-a** [comment]
Simple and Fast Notes but i do not change any file except the itemView component the others changes come automatic by eslint or prettier for styling the code to be more readable

**Review by DevAbdoTolba — CHANGES_REQUESTED** [review]
Almost done. last change _i hope_

**Review by DevAbdoTolba — APPROVED** [review]
LGTM 👍

**Pull request merged**